src/main/java/org/jabref/logic/importer/util/MetaDataParser.java

```diff
@@ -49,61 +51,55 @@ public MetaData parse(MetaData metaData, Map<String, String> data, Character key
         List<String> defaultCiteKeyPattern = new ArrayList<>();
         Map<EntryType, List<String>> nonDefaultCiteKeyPatterns = new HashMap<>();
 
-        for (Map.Entry<String, String> entry : data.entrySet()) {
+        // process GROUPSTREE and GROUPSTREE_LEGACY at the very end (otherwise it can happen that not all dependent data are set)
+        Stream<Map.Entry<String, String>> stream = data.entrySet().stream().filter(entry -> !entry.getKey().equals(MetaData.GROUPSTREE) && !entry.getKey().equals(MetaData.GROUPSTREE_LEGACY));
+        Stream<Map.Entry<String, String>> streamTail = data.entrySet().stream().filter(entry -> entry.getKey().equals(MetaData.GROUPSTREE) || entry.getKey().equals(MetaData.GROUPSTREE_LEGACY));
+        stream = Stream.concat(stream, streamTail);
```

Comment:
Can you please use more meaningful variable names? Like `entrySetStream` and `groupsTreeStream` or similar.